projects/kit/components/input-date-range/input-date-range.component.ts

```diff
@@ -157,6 +168,20 @@ export class TuiInputDateRangeComponent
         super(control, cdr, valueTransformer);
     }
 
+    @HostListener('click')
+    onClick(): void {
+        if (!this.isMobile && this.interactive) {
+            this.toggle();
+        }
+    }
+
+    ngAfterViewChecked(): void {
+        if (this.calendarRange) {
+            this.calendarRange.selectedActivePeriod = this.selectedActivePeriod;
+            this.calendarRange.checkUpdate();
+        }
```

Comment:
when calendar rerender pass saved `selectedActivePeriod` to child calendarRange to display correct active item